selfdrive/car/subaru/subarucan.py

```diff
@@ -23,6 +23,28 @@ def create_steering_control(packer, car_fingerprint, apply_steer, frame, steer_s
     }
     values["Checksum"] = subaru_checksum(packer, values, 0x122)
 
+  if car_fingerprint in (CAR.OUTBACK, CAR.LEGACY):
+
+    if apply_steer != 0:
+      chksm_steer = apply_steer * -1
+      chksm_engage = 1
+    else:
+      chksm_steer = 0
+      chksm_engage = 0
+
+    #counts from 0 to 7 then back to 0
+    idx = (frame / steer_step) % 8
+    steer2 = (chksm_steer >> 8) & 0x1F
+    steer1 =  chksm_steer - (steer2 << 8)
+    checksum = (idx + steer2 + steer1 + chksm_engage) % 256
```

Comment:
I would define a generic "subaru_checksum_legacy" function. It must be computed in some generic way...

Reply:
I'll try, it's just the sum of all bytes mod 256.
I need to do it anyway because the majority of the messages use the same checksum including the button control message and I need that for cancelling without using doors and for stop+go auto-resume